SpinalHDL errors when adding --debug-jtag-tap for JTAG debugging

I want to debug code running on VexiiRiscv soft-core on my Xilinx Nexys A7-100T FPGA (Atrix-7) using JTAG (GDB + OpenOCD)

After adding the --debug-jtag-tap option as you suggested in the docs, I got the following errors:
SCRIPT:
```
sbt "Test/runMain vexiiriscv.Generate
	--xlen 64 \
	--reset-vector 0x00004000 \
	--with-mul --with-div --with-rva --with-rvc \
	--with-rvZcbm \
	--with-supervisor --with-user \
	--allow-bypass-from=0 --performance-counters=0 \
	--lsu-l1 --lsu-axi4 --lsu-l1-axi4 --lsu-l1-ways=4 --lsu-l1-mem-data-width-min=64 --with-lsu-bypass \
	--fetch-l1 --fetch-axi4 --fetch-l1-ways=4 --fetch-l1-mem-data-width-min=64 \
	--with-btb --with-ras --with-gshare \
	--with-rdtime \
	--debug-jtag-tap \
	--region base=00000000,size=F000000,main=1,exe=1 \
	--region base=80000000,size=8000000,main=1,exe=1 \
	--region base=10000000,size=6F000000,main=0,exe=0"
```

ERRORS:
```
./scripts/generate_cpu.sh
[info] welcome to sbt 1.10.0 (Debian Java 21.0.9)
[info] loading settings for project vexiiriscv-build from plugins.sbt ...
[info] loading project definition from /home/zakaria/devspace/rive-fpga/submodules/VexiiRiscv/project
[info] loading settings for project ret from build.sbt ...
[info] loading settings for project spinalhdl-build from plugin.sbt ...
[info] loading project definition from /home/zakaria/devspace/rive-fpga/submodules/VexiiRiscv/ext/SpinalHDL/project
[info] loading settings for project all from build.sbt ...
[info] set current project to VexiiRiscv (in build file:/home/zakaria/devspace/rive-fpga/submodules/VexiiRiscv/)
[info] running (fork) vexiiriscv.Generate --xlen 64 --reset-vector 0x00004000 --with-mul --with-div --with-rva --with-rvc --with-rvZcbm --with-supervisor --with-user --allow-bypass-from=0 --performance-counters=0 --lsu-l1 --lsu-axi4 --lsu-l1-axi4 --lsu-l1-ways=4 --lsu-l1-mem-data-width-min=64 --with-lsu-bypass --fetch-l1 --fetch-axi4 --fetch-l1-ways=4 --fetch-l1-mem-data-width-min=64 --with-btb --with-ras --with-gshare --with-rdtime --debug-jtag-tap --region base=00000000,size=F000000,main=1,exe=1 --region base=80000000,size=8000000,main=1,exe=1 --region base=10000000,size=6F000000,main=0,exe=0
[info] [Runtime] SpinalHDL dev    git head : 84ef7b5b6278854e0e80738536f72c64fa14b730
[info] [Runtime] JVM max memory : 3948.0MiB
[info] [Runtime] Current date : 2026.01.09 19:59:14
[info] [Progress] at 0.000 : Elaborate components
[info] ERROR !
[info] A null pointer access has been detected in the JVM.
[info] This could happen when in your SpinalHDL description, you access an signal which is only defined further.
[info] For instance :
[info]   val result = Bool()
[info]   result := a ^ b  //a and b can't be accessed there because they are only defined one line below (Software rule of execution order)
[info]   val a,b = Bool()
[info]           
[error] Exception in thread "main" java.lang.NullPointerException: Cannot invoke "Object.getClass()" because "qual3" is null
[error]         at vexiiriscv.misc.EmbeddedRiscvJtag$$anon$1$$anon$2.<init>(EmbeddedJtagPlugin.scala:75)
[error]         at vexiiriscv.misc.EmbeddedRiscvJtag$$anon$1.$anonfun$onDebugCd$1(EmbeddedJtagPlugin.scala:31)
[error]         at spinal.core.ClockDomain.apply(ClockDomain.scala:431)
[error]         at spinal.core.ClockDomain.on(ClockDomain.scala:436)
[error]         at vexiiriscv.misc.EmbeddedRiscvJtag$$anon$1.<init>(EmbeddedJtagPlugin.scala:31)
[error]         at vexiiriscv.misc.EmbeddedRiscvJtag.$anonfun$logic$1(EmbeddedJtagPlugin.scala:26)
[error]         at spinal.lib.misc.plugin.FiberPlugin$$anon$1.$anonfun$build$4(Fiber.scala:87)
[error]         at spinal.core.ScopeProperty$ApplyClass.apply(ScopeProperty.scala:144)
[error]         at spinal.lib.misc.plugin.PluginHost.rework(Host.scala:29)
[error]         at spinal.lib.misc.plugin.FiberPlugin$$anon$1.$anonfun$build$2(Fiber.scala:85)
[error]         at spinal.core.internals.BooleanPimped.generate(Misc.scala:285)
[error]         at spinal.lib.misc.plugin.FiberPlugin$$anon$1.$anonfun$build$1(Fiber.scala:82)
[error]         at spinal.core.fiber.Fiber.$anonfun$addTask$1(Fiber.scala:74)
[error]         at spinal.core.fiber.package$.$anonfun$hardForkRawHandle$1(package.scala:30)
[error]         at spinal.core.fiber.AsyncThread.$anonfun$jvmThread$1(AsyncThread.scala:60)
[error]         at spinal.core.fiber.EngineContext.$anonfun$newJvmThread$1(AsyncCtrl.scala:39)
[error]         at spinal.sim.JvmThread.run(SimManager.scala:51)
[error] Nonzero exit code returned from runner: 1
[error] (Test / runMain) Nonzero exit code returned from runner: 1
[error] Total time: 3 s, completed Jan 9, 2026, 7:59:15 PM
make: *** [Makefile:44: cpu] Error 1
```

I also would like to ask, how can I connect the JTAG pins exposed by the VexiiRiscv Verilog module to the USB-JTAG pins on my FPGA?

Because they are primary used for programming the FPGA though /dev/ttyUSB0 used by Vivado, and the [Digilent provided .xdc file for my FPGA](https://github.com/Digilent/digilent-xdc/blob/master/Nexys-A7-100T-Master.xdc) does not expose any JTAG pins.

--- a/src/main/scala/vexiiriscv/Param.scala
+++ b/src/main/scala/vexiiriscv/Param.scala
@@ -664,8 +664,8 @@ class ParamSimple() {
     opt[Unit]("debug-privileged") action { (v, c) => privParam.withDebug = true }
     opt[Int] ("debug-triggers") action { (v, c) => privParam.debugTriggers = v }
     opt[Unit]("debug-triggers-lsu") action { (v, c) => privParam.debugTriggersLsu = true }
-    opt[Unit]("debug-jtag-tap") action { (v, c) => embeddedJtagTap = true }
-    opt[Unit]("debug-jtag-instruction") action { (v, c) => embeddedJtagInstruction = true }
+    opt[Unit]("debug-jtag-tap") action { (v, c) => embeddedJtagTap = true; privParam.withDebug = true }
+    opt[Unit]("debug-jtag-instruction") action { (v, c) => embeddedJtagInstruction = true; privParam.withDebug = true }
     opt[Unit]("with-boot-mem-init") action { (v, c) => bootMemClear = true }
     opt[Int]("physical-width") action { (v, c) => physicalWidth = v }
     opt[Unit]("mul-keep-src") action { (v, c) => mulKeepSrc = true }

